src/main/java/org/lsc/configuration/JaxbXmlConfigurationHelper.java

```diff
@@ -113,56 +119,102 @@ public JaxbXmlConfigurationHelper() throws LscConfigurationException {
-            return (Lsc)unmarshaller.unmarshal( new File(filename) );
+            return (Lsc)unmarshaller.unmarshal(IOUtils.toInputStream(xmlContentWithEnvInlined));
         } catch (JAXBException e) {
             throw new LscConfigurationException(e);
         }
     }
 
+	private Unmarshaller getUnmarshallerWithSchemaValidation() throws JAXBException, LscConfigurationException {
+		Unmarshaller unmarshaller = jaxbc.createUnmarshaller();
+		SchemaFactory schemaFactory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
+		Schema lscSchema = null;
+		try {
+		    int i = 0;
+		    Set<URL> urls = new HashSet<URL>();
+		    urls.addAll(ClasspathHelper.forPackage("org.lsc"));
+		    if(System.getProperty("LSC.PLUGINS.PACKAGEPATH") != null) {
+		        String[] pathElements = System.getProperty("LSC.PLUGINS.PACKAGEPATH").split(System.getProperty("path.separator"));
+		        for(String pathElement: pathElements) {
+		            urls.addAll(ClasspathHelper.forPackage(pathElement));
+		        }
+		    }
+		    Reflections reflections = new Reflections(new ConfigurationBuilder()
+		        .addUrls(urls).setScanners(new ResourcesScanner(), new SubTypesScanner()));
+		    
+		    Set<String> xsdFiles = reflections.getResources(Pattern.compile(".*\\.xsd"));
+		    Source[] schemasSource = new Source[xsdFiles.size()];
+		    List<String> xsdFilesList = new ArrayList<String>(xsdFiles);
+		    Collections.sort(xsdFilesList, new XsdForLscComparator());
+		    for(String schemaFile: xsdFilesList) {
+		        LOGGER.debug("Importing XML schema file: " + schemaFile);
+		        InputStream schemaStream = this.getClass().getClassLoader().getResourceAsStream(schemaFile);
+		        schemasSource[i++] = new StreamSource(schemaStream);
+		    }
+		    lscSchema = schemaFactory.newSchema(schemasSource);
+		    unmarshaller.setSchema( lscSchema );
+		    
+		} catch (VerifyError e) {
+		    throw new LscConfigurationException(e.toString(), e);
+		} catch (SAXException e) {
+		    throw new LscConfigurationException(e);
+		}
+		return unmarshaller;
+	}
+	
+	private String getEnvInlinedContent(Map<String, String> env, String filename) throws LscConfigurationException{
+		BufferedReader reader = null;
+		try {
+			reader = new BufferedReader(new FileReader(filename));
+			String line = reader.readLine();
+			StringWriter writer = new StringWriter();
+
+			while (line != null) {
+				writer.write(inlineEnvForLine(env, line) + System.lineSeparator());
+				line = reader.readLine();
+			}
+			return writer.toString();
+		} catch (Exception e) {
+			throw new LscConfigurationException(e);
+		} finally {
+			if (reader != null) {
+				try {
+					reader.close();
+				} catch (IOException e) {
+					LOGGER.error("Error during closing the configuration file " + filename, e);
+				}
+			}
+		}
+	}
+	
+	private String inlineEnvForLine(Map<String, String> env, String line) {
+		String resultString = line;
+		
+		for(String envVariableKey : env.keySet()) {
```

Comment:
can you quickly check if doing this for every env variable does not slow down parsing xml files too much? If this is the case we could add a little optimization checking before if the line contains `${`

Reply:
It seems to not have much of an inpact on the 232 lines xml test file.
i have an average time for 100 runs of loading the file of : 
- 178 ms for the file without environment variable placeholder and no environment variable provided.
- 180 ms for the file with environment variable placeholder and 110 environment variables.